packages/blockchain-wallet-v4-frontend/src/scenes/InterestHistory/DownloadTransactions/index.tsx

```diff
@@ -0,0 +1,176 @@
+
+    if (prevProps.formValues?.coin !== formValues?.coin) {
+      interestActions.clearInterestTransactionsReport()
+      this.setState({ hasSavedReport: false }) // eslint-disable-line
+    }
+  }
+
+  componentWillUnmount() {
+    this.props.interestActions.clearInterestTransactionsReport()
+  }
+
+  handleDownload = () => {
+    const { interestActions } = this.props
+    interestActions.fetchInterestTransactionsReport()
+  }
+
+  handleSaveReport = () => {
+    this.setState({ hasSavedReport: true })
+  }
+
+  render() {
+    const { transactionsReportR, txPages } = this.props
+    const txList = flatten(
+      txPages &&
+        // @ts-ignore
+        txPages.map(pages => map(page => page, (pages && pages.data) || []))
+    )
+
+    return (
+      txList?.length > 0 &&
+      transactionsReportR.cata({
+        Success: val => {
+          return this.state.hasSavedReport ? (
+            <SuccessIconButton
+              data-e2e='interestTxReportSaved'
+              disabled
+              height='45px'
+              name='checkmark-in-circle-filled'
+              nature='primary'
+              width='140px'
+            >
+              <FormattedMessage
+                id='scenes.interest.transactions.saved'
+                defaultMessage='Saved'
+              />
+            </SuccessIconButton>
+          ) : (
+            <DownloadButton
+              data={val}
+              filename='Interest_Transactions.csv'
+              onClick={this.handleSaveReport}
+              target='_blank'
+              width='100%'
+            >
+              <StyledIconButton
+                data-e2e='saveInterestTxReport'
+                height='45px'
+                name='download'
+                nature='light'
+                width='140px'
+              >
+                <FormattedMessage
+                  id='scenes.interest.transactions.savereport'
+                  defaultMessage='Save Report'
+                />
+              </StyledIconButton>
+            </DownloadButton>
+          )
+        },
+        Failure: () => (
+          <FailedIconButton
+            data-e2e='interestTxReportFailed'
+            height='45px'
+            name='alert-filled'
+            nature='primary'
+            width='140px'
+          >
+            <FormattedMessage id='copy.failed' defaultMessage='Failed' />
+          </FailedIconButton>
+        ),
+        Loading: () => (
+          <LoaderContainer>
+            <HeartbeatLoader />
+          </LoaderContainer>
+        ),
+        NotAsked: () => (
+          <StyledIconButton
+            data-e2e='generateInterestTxReport'
+            height='45px'
+            onClick={this.handleDownload}
+            name='download'
+            nature='light'
+            width='140px'
+          >
+            <FormattedMessage
+              id='scenes.transactions.export.download'
+              defaultMessage='Download'
```

Comment:
`copy.download`?

Reply:
There is not a copy.download, should I change all the other Downloads to copy.download?